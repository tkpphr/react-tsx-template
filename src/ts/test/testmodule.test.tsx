import * as React from 'react';
import { mount } from 'enzyme';
import mountToJson from "enzyme-to-json";
import TestModule from "../main/testmodule";

describe('Props Test', () => {
    test('<TestModule>', () => {
        const elem = mount(<TestModule content="click" />);
        elem.find('button').simulate('click');
        let currentDom = mountToJson(elem);
        expect(currentDom).toMatchSnapshot();
    
        elem.find('button').simulate('click');
        currentDom = mountToJson(elem);
        expect(currentDom).toMatchSnapshot();
    });
});