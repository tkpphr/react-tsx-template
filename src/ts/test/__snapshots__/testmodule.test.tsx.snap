// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Props Test <TestModule> 1`] = `
<TestModule
  content="click"
>
  <styled.div>
    <div
      className="sc-bdVaJa bBqccL"
    >
      <styled.button
        onClick={[Function]}
      >
        <button
          className="sc-bwzfXH dfMRuM"
          onClick={[Function]}
        >
          click
        </button>
      </styled.button>
      <styled.p>
        <p
          className="sc-htpNat jmicNh"
        >
          Clicked : 
          1
        </p>
      </styled.p>
    </div>
  </styled.div>
</TestModule>
`;

exports[`Props Test <TestModule> 2`] = `
<TestModule
  content="click"
>
  <styled.div>
    <div
      className="sc-bdVaJa bBqccL"
    >
      <styled.button
        onClick={[Function]}
      >
        <button
          className="sc-bwzfXH dfMRuM"
          onClick={[Function]}
        >
          click
        </button>
      </styled.button>
      <styled.p>
        <p
          className="sc-htpNat jmicNh"
        >
          Clicked : 
          2
        </p>
      </styled.p>
    </div>
  </styled.div>
</TestModule>
`;